add_me_to_your_PATH/Dockerfile.template

```diff
@@ -149,7 +149,7 @@ RUN --mount=type=cache,sharing=locked,target=/tmp/julia-cache \
 # Install required dependencies for sysimage.jl
 RUN --mount=type=cache,sharing=locked,target=/tmp/julia-cache \
     --mount=type=secret,id=github_token \
-    julia -e 'using Pkg; Pkg.add(PackageSpec(name="PackageCompiler", version="1"); preserve=Pkg.PRESERVE_ALL)'
+    julia -e 'using Pkg; Pkg.activate(); Pkg.add(PackageSpec(name="PackageCompiler", version="1"); preserve=Pkg.PRESERVE_ALL)'
```

Comment:
> If no argument is given to activate, then activate the home project. The
> home project is specified by either the `--project` command line option to the
> julia executable, or the `JULIA_PROJECT` environment variable.

As we specify a `JULIA_PROJECT` in this stage then this should be a no-op.

I think we need to update the Dockerfile test to reproduce the scenario you're seeing

Reply:
I tried this with JULIA_PROJECT set and it still activated the global env:

```
➜ JULIA_PROJECT=~/.julia/dev/Example julia
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.9.0 (2023-05-07)
 _/ |\__'_|_|_|\__'_|  |  Official https://julialang.org/ release
|__/                   |

julia> using Pkg

julia> Pkg.project().path
"/Users/dkleinschmidt/.julia/dev/Example/Project.toml"

julia> ENV["JULIA_PROJECT"]
"/Users/dkleinschmidt/.julia/dev/Example"

julia> Pkg.activate()
  Activating project at `~/.julia/environments/v1.9`

julia> Pkg.project().path
"/Users/dkleinschmidt/.julia/environments/v1.9/Project.toml"
```

Reply:
The plan is to use `JULIA_PROJECT=""` instead of this as it should continue to work even if `Pkg.activate`'s behavior changes